win32ss/user/ntuser/winpos.c

```diff
@@ -1393,8 +1393,28 @@ WinPosDoOwnedPopups(PWND Window, HWND hWndInsertAfter)
 
                if (List[i] == Owner)
                {
-                  if (i > 0) hWndInsertAfter = List[i-1];
-                  else hWndInsertAfter = topmost ? HWND_TOPMOST : HWND_TOP;
+                  /* We found its Owner, so we must handle it here. */
+                  if (i > 0)
+                  {
+                     if (List[i - 1] != Window->head.h)
+                     {
+                        /* Do not allow hWndInsertAfter to become equal to
+                         * Window->head.h. This would cause the window to
+                         * reference itself. This changes the passed in
+                         * hWndInsertAfter which will be handled below. */
```

Comment:
```suggestion
                        /*
                         * If the popup to be inserted is not already just
                         * before the Owner, insert it there. The modified
                         * hWndInsertAfter will be handled below.
                         *
                         * (NOTE: Do not allow hWndInsertAfter to become equal
                         * to the popup's window handle, as this would cause
                         * the popup to link to itself).
                         */
```

Reply:
Changed. Nice comments. Thanks.